test/functional/wallet_dump.py

```diff
@@ -47,8 +47,14 @@ def read_dump(file_name, addrs, script_addrs, hd_master_addr_old):
 
                     # count key types
                     for addrObj in addrs:
-                        if addrObj['address'] == addr and addrObj['hdkeypath'] == keypath and keytype == "label=":
-                            found_addr += 1
+                        if addrObj['address'] == addr.split(",")[0] and addrObj['hdkeypath'] == keypath and keytype == "label=":
+                            # ensure we only add the length of the set of addresses in an entry
+                            # containing multiple addresses if it is the expected p2wsh-p2wpkh
+                            # we added in the test
+                            if len(set(addr.split(","))) > 1 and addr.split(",")[1] == script_addrs[1]:
+                                found_addr += len(set(addr.split(",")))
```

Comment:
this line seems to be unreached when this test is run

Reply:
Good catch :) the test was passing for because I broke it.  I'll set the test to check for the p2sh-p2wpkh address specifically for the key it was added to.